geoips/commandline/geoips_command.py

```diff
@@ -303,59 +331,192 @@ def _get_registry_by_interface_and_package(self, interface, package_name):
+        return table_data
+
+    def _get_entry(self, plugin_dict, headers):
+        """Retrieve the appropriate plugin entry given a list of valid headers.
+
+        Parameters
+        ----------
+        plugin_dict: dict
+            - The portion of the plugin_registry needed to access the appropriate plugin
+        headers: dict
+            - Dictionary of strings representing the key-value mapping of the
+              headers we'd like to output.
+
+        Returns
+        -------
+        plugin_entry: list of str
+            - A list of strings containing information about a plugin listed in the
+              order of headers.keys()
+        """
+        table_data = []
+        for plugin_key in list(plugin_dict.keys()):
+            plugin_entry = []
+            for header in list(headers.keys()):
+                if (
+                    header == "source_names"
+                    and plugin_dict[plugin_key]["interface"] != "products"
+                ):
+                    plugin_entry.append("N/A")
+                elif (
+                    header == "family"
+                    and plugin_dict[plugin_key]["interface"] == "products"
+                ):
+                    plugin_entry.append("N/A")
+                elif header == "plugin_name":
+                    plugin_entry.append(plugin_key)
+                else:
+                    plugin_entry.append(plugin_dict[plugin_key][header])
+            table_data.append(plugin_entry)
+        return table_data
+
+
+class GeoipsListCommon(GeoipsExecutableCommand):
+    """Class containing common optional arguments shared between list commands."""
+
+    command_name = "list_common"
+    subcommand_classes = []
+
+    def add_arguments(self):
+        """Add arguments to the list-subparser for the List Command."""
+        self.subcommand_parser.add_argument(
+            "--package_name",
+            "-p",
+            type=str,
+            default="all",
+            choices=self.plugin_package_names,
+            help="The GeoIPS package to list from.",
+        )
+        mutex_group = self.subcommand_parser.add_mutually_exclusive_group()
+        mutex_group.add_argument(
+            "--long",
+            "-l",
+            default=True,
+            action="store_true",
+            help="Flag representing the 'long' listing of a certain command.",
+        )
+        mutex_group.add_argument(
+            "--columns",
+            "-c",
+            type=str,
+            nargs="+",
+            default=None,
+            help="""Specific Headers of Data you'd like to see listed.
+                    For more in formation on headers available, run
+                    'geoips list <sub-cmd> <positional_args> --columns help'.
+                    """,
+        )
+
+    def __call__(self, args):
+        """Exectutable function for GeoipsListCommon Class. Not implemented."""
+        pass
```

Comment:
Why is this an executable command?

Reply:
Executable commands have the ability to add arguments. Since we want shared arguments between all of the list commands, this was the only way to do this via our Abstract Class and argparse's implementation of sharing arguments.

Reply:
Okay, this is fine for now. I think we'll probably discuss this more when we discuss the inheritance structure as @biosafetylvl5 has suggested in the past. I'll just resolve this for now since it will get handled in the future, one way or another.